### ISSUE TITLE
Hello - togetherjs not working

### ISSUE THREAD

**Issue by stevecalla** [mozilla/togetherjs#1197]
Hello - I installed togetherjs as instructed but not working. Invite sent to another person. They were asked to join but after that nothing seems to happen to connect. Here's the site. https://stevecalla.github.io/solo-project-slap-jack/#. Maybe I missed a step (I added the script line above body and added the button as you can see on the page). Steve

**Comment by mamamia5x** [comment]
I think it's been not working forever now. I've tried to get this project to work. And I've used it for projects before, and it worked, but now it doesn't. I don't really understand,  because at one point it was working just fine. Then, it just stopped. But, I've tried using it on the [site](https://togetherjs.com/) itself, and still no luck. I hope that it's just some bug, because this project is really cool. I don't know if anyone else is experiencing this either, but hopefully a solution can be found (if infact it is a bug).

**Comment by stevesap** [comment]
@stevecalla your application seems to have the issue:

`togetherjsPackage.js?bust=1612520151841:1272 WebSocket connection to 'wss://hub.togetherjs.com/hub/vTqeudTHQx' failed: Error in connection establishment: net::ERR_NAME_NOT_RESOLVED
`

could it be due to the server being down? (see #1172)

**Comment by mamamia5x** [comment]
That probably explains it, and I've had this issue since around this time. But this isn't really anything we can do to fix it.

**Comment by stevecalla** [comment]
@stevesap @mamamia5x Thanks so much!

**Comment by biologistbrian** [comment]
No it works! definitely use #1172 

There's a great post on deploying a glitch server.  It's magic!

**Comment by zhu956645239** [comment]
@stevesap @mamamia5x  Have you ever had this problem？？
togetherjsPackage.js?bust=1615541581728:1955 Uncaught TypeError: msg.peer.updateFromHello is not a function
    at constructor.c.onmessage (togetherjsPackage.js?bust=1615541581728:1955)
    at constructor._incoming (togetherjsPackage.js?bust=1615541581728:1261)
    at constructor.<anonymous> (togetherjsPackage.js?bust=1615541581728:1370)

**Comment by mamamia5x** [comment]
No, I get an error that the Websocket cannot connect to the server.  
channels.js?bust=1615556464444:174 WebSocket close unclean code: 1006 reason: none
1006 Error means that the connection closed locally. 
Like @biologistbrian said, you just need to create a server.

**Comment by zedttxj** [comment]
Hi,
Just a heads-up: the original TogetherJS hub server is currently down. I’ve deployed my own on Render (free tier), and it works fine—but you’ll need to adjust server.js (from the hub folder) a bit to get it running if you wanna deploy for public project: change the process.env.HOST || '127.0.0.1'; into process.env.HOST || '0.0.0.0';

The only outdated tool here is the voice chat feature. You gotta manually create the grunt package and make it public (on GitHub or somewhere else). After creating it, go to togetherjs/togetherjsPackage.js and change the element.src = URL.createObject... of the function attachMedia into element.srcObject = media. In the function startStreaming, change the $video[0].src = URL.createObject... into $video[0].srcObject = stream. Basically, we replace all the URL.createObject and replace the src into srcObject. Optionally, the makePeerConnection only points to STUN server. Sometimes in school networks the server doesn't work so we add TURN server (ideally your own). But, if you wanna deploy locally, you wouldn't need it. Here's the free version of TURN server:
function makePeerConnection() {
    const config = {
      iceServers: [
        { urls: "stun:stun.l.google.com:19302" },
        {
          urls: "turn:openrelay.metered.ca:80",
          username: "openrelayproject",
          credential: "openrelayproject"
        }
      ]
    };

    const constraints = {
      optional: [{ DtlsSrtpKeyAgreement: true }]
    };

    if (window.RTCPeerConnection) {
      return new RTCPeerConnection(config, constraints);
    }

    // For legacy browsers, fallback (rarely needed now)
    if (window.webkitRTCPeerConnection) {
      return new webkitRTCPeerConnection(config, constraints);
    }
    if (window.mozRTCPeerConnection) {
      return new mozRTCPeerConnection(config, constraints);
    }

    throw new Error("No RTCPeerConnection support in this browser.");
  }
Final step to make the voice chat work: set the variable of defaultHubBase (from togetherjs-min.js) to your own hub server. Also, set baseUrl variable into your public togetherjs packages.
